File: 3745.maximize-expression-of-three-elements/solution.go
func maximizeExpressionOfThree(nums []int) int {
    a := math.MinInt
    b := math.MinInt
    c := math.MaxInt
    for _, num := range nums {
        if num > a {
            b = a
            a = num
        } else if num > b {
            b = num
        }
        if num < c {
            c = num
        }
    }
    return a + b - c
}

File: 3745.maximize-expression-of-three-elements/solution.kt
class Solution {
    fun maximizeExpressionOfThree(nums: IntArray): Int {
        var a = Int.MIN_VALUE
        var b = Int.MIN_VALUE
        var c = Int.MAX_VALUE
        nums.forEach { num ->
            if (num > a) {
                b = a
                a = num
            } else {
                b = maxOf(b, num)
            }
            c = minOf(c, num)
        }
        return a + b - c
    }
}